### PR TITLE
Fixed building with AMR disabled.

### PR DESCRIPTION
**Pull request by hrittich**
Building with --disable-amr failed. This should fix it.

### CODE REVIEW

File: contrib/netcdf/4.4.1.1/include/netcdf_meta.h [hunk 1]
```diff
@@ -38,10 +38,10 @@
 #define NC_VERSION       "4.4.1.1"
 
 #define NC_HAS_NC2       1 /*!< API version 2 support. */
-#define NC_HAS_NC4       1 /*!< API version 4 support. */
+#define NC_HAS_NC4       0 /*!< API version 4 support. */
 #define NC_HAS_HDF4      0 /*!< hdf4 support. */
-#define NC_HAS_HDF5      1 /*!< hdf5 support. */
-#define NC_HAS_SZIP      1 /*!< szip support (hdf5 only) */
```
Comment (roystgnr) [inline contrib/netcdf/4.4.1.1/include/netcdf_meta.h]: How are these three changes necessary?

File: include/numerics/dense_vector.h [hunk 2]
```diff
@@ -28,7 +28,7 @@
 
 #ifdef LIBMESH_HAVE_EIGEN
 #include "libmesh/ignore_warnings.h"
-#include <Eigen/Dense>
+#include <Eigen/Core>
 #include "libmesh/restore_warnings.h"
```
Comment (roystgnr) [inline include/numerics/dense_vector.h]: This looks like a correct change, but probably belongs split into a separate PR to make bisecting easier if people encounter problems - surely including too much from Eigen wasn't causing --disable-amr problems?

File: include/numerics/eigen_core_support.h [hunk 3]
```diff
@@ -45,7 +45,7 @@
 #include "libmesh/ignore_warnings.h"
 
 // Eigen includes
-#include <Eigen/Dense>
```
Comment (roystgnr) [inline include/numerics/eigen_core_support.h]: This, on the other hand, I'm not so sure about.  radial_basis_interpolation.C uses Eigen dense solvers, which aren't in Core, are they?

File: src/solution_transfer/radial_basis_interpolation.C [hunk 4]
```diff
@@ -27,6 +27,9 @@
 #include "libmesh/libmesh_logging.h"
 #include "libmesh/eigen_core_support.h"
 
+#ifdef LIBMESH_HAVE_EIGEN
+#include <Eigen/Dense>
+#endif
```
Comment (roystgnr) [inline src/solution_transfer/radial_basis_interpolation.C]: Ah, now I see.  Disregard what I said about the eigen_core_support.h change above.

File: include/numerics/eigen_core_support.h (hunk 3 above)
Comment (roystgnr) [inline include/numerics/eigen_core_support.h]: Never mind, just saw the corresponding change there.

**Comment by roystgnr** [comment]
The CheckpointIO fixes look good; sorry for messing that up for you!

The Eigen changes look good, but should probably be split into a separate PR, unless they really do fix something specific to --disable-amr.

The NetCDF changes I don't get.

**Comment by hrittich** [comment]
Sorry. I messed up the commit.

Those should have been two separate things...

**Comment by hrittich** [comment]
This should be it, to fix the --disable-amr.

**Comment by roystgnr** [comment]
:+1: once the tests pass.. and probably even if they fail; there's no way these changes should trigger anything that isn't a false positive.

**Comment by brianmoose** [comment]
FYI, there is already an optional CIVET test for this, which I added to this PR.

**Comment by jwpeterson** [comment]
Oh, I see, this was just recently broken in 314e09c473. Thanks @brianmoose for attaching the `--disable-amr` test.
